src/main/java/net/neoforged/neoforge/client/event/RegisterDimensionTransitionScreenEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.BooleanSupplier;
+import net.minecraft.client.gui.screens.ReceivingLevelScreen;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.Level;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.fml.event.IModBusEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * <p>Event for registering screen effects when transitioning across dimensions.
+ * Note that there is a priority order when it comes to what screens are displayed.
+ * If a dimension has a screen that displays when entering it, that will have priority over a dimension that has one when you leave it. </p>
+ *
+ * <p>This event is not {@linkplain ICancellableEvent cancellable}, and does not {@linkplain HasResult have a result}.
+ *
+ * <p>This event is fired on the mod-specific event bus, only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class RegisterDimensionTransitionScreenEvent extends Event implements IModBusEvent {
```

Comment:
Would it be useful to allow registering a transition that only applies when both origin and destination match? It would use a `Pair<RK<Level>, RK<Level>>` as the key and would be checked before the "from effects" and "to effects".

Reply:
I could do that, not sure if anyone would use something like that but it may be nice to have